src/library_egl.js

```diff
@@ -27,16 +49,69 @@ var LibraryEGL = {
         EGL.setErrorCode(0x3008 /* EGL_BAD_DISPLAY */);
         return 0;
       }
-      // TODO: read attribList.
       if ((!config || !config_size) && !numConfigs) {
         EGL.setErrorCode(0x300C /* EGL_BAD_PARAMETER */);
         return 0;
       }
-      if (numConfigs) {
-        {{{ makeSetValue('numConfigs', '0', '1', 'i32') }}}; // Total number of supported configs: 1.
+
+      // config variables to compare against
+      var alpha = 0;
+      var samples = 0;
+      var buffers = 0;
+      var depth = 0;
+      var stencil = 0;
+
+      if (attribList) {
+        while (1) {
+          var param = {{{ makeGetValue('attribList', '0', 'i32') }}};
+          if (param == 0x3038) { // EGL_NONE
+            break;
+          }
+
+          var value = {{{ makeGetValue('attribList', '4', 'i32') }}};
+          switch (param) {
+            case 0x3021: // EGL_ALPHA_SIZE
+              alpha = value;
+              break;
+            case 0x3031: // EGL_SAMPLES
+              samples = value;
+              break;
+            case 0x3032: // EGL_SAMPLE_BUFFERS
+              buffers = value;
+              break;
+            case 0x3025: // EGL_DEPTH_SIZE
+              depth = value;
+              break;
+            case 0x3026: // EGL_STENCIL_SIZE
+              stencil = value;
+              break;
+            default:
+              break;
+          }
+          attribList += 8;
+        }
       }
-      if (config && config_size > 0) {
-        {{{ makeSetValue('config', '0', '62002' /* Magic ID for the only EGLConfig supported by Emscripten */, 'i32') }}}; 
+      var configCount = 0;
+      for (var i=0; i<EGL.configAttribs.length; i++) {
```

Comment:
Space after `=` and `<`, i.e. `for (var i = 0; i < EGL.configAttribs.length; i++) {`.